---
title: Erreur du compilateur CS1631
ms.date: 07/20/2015
f1_keywords:
- CS1631
helpviewer_keywords:
- CS1631
ms.assetid: bf0c5ff9-90a3-4db6-b4ee-0b93e31614e0
ms.openlocfilehash: 7dc7473966b1c78b911c1323b5ea0f8fbf7d8150
ms.sourcegitcommit: 3d5d33f384eeba41b2dff79d096f47ccc8d8f03d
ms.translationtype: HT
ms.contentlocale: fr-FR
ms.lasthandoff: 05/04/2018
ms.locfileid: "33310587"
---
# <a name="compiler-error-cs1631"></a>Erreur du compilateur CS1631
Impossible de générer une valeur dans le corps d'une clause catch  
  
 L’instruction yield n’est pas autorisée à partir du corps d’une clause catch. Pour éviter cette erreur, placez l’instruction yield en dehors du corps de la clause catch.  
  
 L’exemple suivant génère l’erreur CS1631 :  
  
```csharp  
// CS1631.cs  
using System;  
using System.Collections;  
  
public class C : IEnumerable  
{  
   public IEnumerator GetEnumerator()   
   {  
      try  
      {  
      }  
      catch(Exception e)  
      {  
        yield return this;  // CS1631  
      }  
   }    
  
   public static void Main()   
   {  
   }  
}  
```
